### ISSUE TITLE
Create Option To Make Inscriptions Non-Transferable 

### ISSUE THREAD

**Issue by RagnarLifthrasir** [casey/ord#1657]

**Comment by casey** [comment]
Hi Rag! Can you describe a little bit more about what you're looking to get out of this feature?

I have lots of thoughts, but it depends a lot on the use-case. I can't think a good way to make inscriptions non-transferrable. They sit on sats, and sats are transferrable.

If you're looking for something like soul-bound tokens, I have an idea. Making up a use-case: Let's say that you, Rag, are trying to certify people as knowing a lot about 3D-printed firearms. You want to certify me. I have an inscription that I use as my personal identity, CASEYPFP and you have an inscription you use as your personal identity, RAGPFP.

You use RAGPFP (using #783 when implemented) to make an inscription that says, in text "I'm Rag and I saw CASEYPFP field-strip an FGC-9 in 10 seconds." (we can make this better, but that's just a sketch) Only the RAGPFP holder can do this, and it refers to CASEYPFP by the inscription ID, so everyone can see the connection. I can then point people to this inscription.

If you want to revoke it, you send it to a burn address using #1588. You can even, in the OP_RETURN payload, say *why* you're revoking it. Maybe I didn't pay my 3D gun printing club dues, so you just say "nbd, you have to be a member to have this certification, so if casey wants to rejoin he's welcome and we'll make a new one". Or you can say something like "Casey turned out not to actually know anything about 3D-printed firearms, he's a fraud", or whatever.

What do you think?

**Comment by RagnarLifthrasir** [comment]
You described my use case, and know me too well.

Can #783 make a personal identity 1) human readable, 2) globally unique, and strongly owned? I think so? But it might come down to UX and convenience? Basically, I want the properties of the ENS/BNS.

**Comment by RagnarLifthrasir** [comment]
😎 How time flies 📆

<img width="456" alt="Casey Rodarmer DM to Ragnar Lifthrasir - Ordinals and Trajan - December 20 2022" src="https://user-images.githubusercontent.com/24993355/218218399-2403efe2-aab7-4387-83d8-8c97c8120d1b.png">

**Comment by arwyn6969** [comment]
I would like to see non transferrable assets asap if thats a thing.. Scary as it may be. Also note is should be notable if someone intentionally acquired the asset, aka perhaps verify their approval of owning said asset, or also indicting if it was sent to them or otherwise. Seems cool.

**Comment by casey** [comment]
> Can #783 make a personal identity 1) human readable, 2) globally unique, and strongly owned? I think so? But it might come down to UX and convenience? Basically, I want the properties of the ENS/BNS.

Yes! Although it's not #783 that does this, it's #794. #794 will provide counterparty-style globally unique names. I have a weird idea on how to avoid a land grab for short name.

End goal:

- an inscription is your identity
- has a title which is arbitrary text #1005 
- a short unique name provides an easy way to refer to it #794 
- it is owned by one user who can sign to prove they own it using BIP-322 to prove they could spend the UTXO: https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki

**Comment by casey** [comment]
> I would like to see non transferrable assets asap if thats a thing.. Scary as it may be. Also note is should be notable if someone intentionally acquired the asset, aka perhaps verify their approval of owning said asset, or also indicting if it was sent to them or otherwise. Seems cool.

Unfortunately there is no asap at the moment. Everyone is encouraged to come up with ad-hoc solutions to their problems, but the main project needs to do things right and implement things that will be good for the long haul.

**Comment by RagnarLifthrasir** [comment]
@casey What if I want to make an Inscription non-transferable by default? Also, @john-light had an idea, he tweeted, 

_I don't see why an inscription couldn't be appended with a label like "NT" which would make it so that inscription software will not honor/recognize attempts to transfer it after it's been created 🤔_

**Comment by tyjvazum** [comment]
@RagnarLifthrasir, thank you for your discussion on this feature. I implemented support for a version of the concept in [arb](https://github.com/tyjvazum/arb). Any improvements you have in mind would be very welcome 👍
